### DISCUSSION TITLE
Show turn by turn instructions in GraphHopper 4.0

### DISCUSSION THREAD

**Discussion by huseynehedov** [mapsforge/vtm#964] [Q&A] [answered]
How to show instructions turn by turn? I have route and I want to set up the routing if I change my location, then the set up route again from the graphopper and show instructions.
For example: turn right or keep right etc. Thanks for advanced.

**Comment by devemux86** [comment, marked as answer]
VTM is an OpenGL map library.
To show the location see the [LocationActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/LocationActivity.java) and [LocationTextureActivity](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/LocationTextureActivity.java) samples.

---

The question is more about how to write a navigator app.
